src/rigid_optimization/sdf_2_sdf_optimizer2d.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * sdf2sdf_optimizer2d.cpp
+ *
+ *  Created on: Mar 05, 2019
+ *      Author: Fei Shan
+ */
+
+//libraries
+#include <boost/python.hpp>
+#include <iostream>
+
+// local
+#include "../math/transformation.hpp"
+#include "sdf_2_sdf_optimizer2d.hpp"
+#include "sdf_gradient_wrt_transformation2d.hpp"
+#include "../tsdf/tsdf.hpp"
+
+namespace ropt = rigid_optimization;
+namespace eig = Eigen;
+
+namespace rigid_optimization {
+
+Sdf2SdfOptimizer2d::Sdf2SdfOptimizer2d(
+        float rate,
+        int maximum_iteration_count,
+        VerbosityParameters verbosity_parameters) :
+                rate(rate),
+                maximum_iteration_count(maximum_iteration_count),
+                verbosity_parameters(verbosity_parameters)
+{
+};
+
+Sdf2SdfOptimizer2d::~Sdf2SdfOptimizer2d()
+{
+};
+
+Sdf2SdfOptimizer2d::VerbosityParameters::VerbosityParameters(
+        bool print_iteration_max_warp_update,
+        bool print_iteration_energy) :
+            print_iteration_max_warp_update(print_iteration_max_warp_update),
+            print_iteration_energy(print_iteration_energy),
+            print_per_iteration_info(
+                    print_iteration_max_warp_update ||
+                    print_iteration_energy
+            )
+{
+};
+
+Sdf2SdfOptimizer2d::TSDFGenerationParameters::TSDFGenerationParameters(
+        float depth_unit_ratio,
+        eig::Matrix3f camera_intrinsic_matrix,
+        eig::Matrix4f camera_pose,
+        eig::Vector3i array_offset,
+        int field_size,
+        float voxel_size,
+        int narrow_band_width_voxels) :
+            depth_unit_ratio(depth_unit_ratio),
+            camera_intrinsic_matrix(camera_intrinsic_matrix),
+            camera_pose(camera_pose),
+            array_offset(array_offset),
+            field_size(field_size),
+            voxel_size(voxel_size),
+            narrow_band_width_voxels(narrow_band_width_voxels)
+{
+};
+
+eig::Vector3f Sdf2SdfOptimizer2d::optimize(int image_y_coordinate,
+        const eig::Matrix<unsigned short, eig::Dynamic, eig::Dynamic>& canonical_depth_image,
+        const eig::Matrix<unsigned short, eig::Dynamic, eig::Dynamic>& live_depth_image,
+        const Sdf2SdfOptimizer2d::TSDFGenerationParameters tsdf_generation_parameters,
+        float eta = 0.01f) {
```

Comment:
Default arguments should be defined in the header and only in the header, not the .cpp file.

Reply:
fixed